public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -199,9 +202,47 @@ export function recordingRulerRuleToRuleForm(
   };
 }
 
-export const getDefaultQueries = (): AlertQuery[] => {
+export const getDefaultQueriesAsync = async (): Promise<{
+  queries: AlertQuery[] | null;
+  ds?: DataSourceApi<DataQuery, DataSourceJsonData, {}>;
+}> => {
   const dataSource = getDefaultOrFirstCompatibleDataSource();
 
+  if (!dataSource) {
+    return { queries: [...getDefaultExpressions('A', 'B')] };
+  }
+  const relativeTimeRange = getDefaultRelativeTimeRange();
+
+  let ds;
+  try {
+    ds = await getDataSourceSrv().get(dataSource.uid);
+  } catch (error) {
+    console.log(error);
+    return { queries: [...getDefaultExpressions('A', 'B')] };
+  }
+
+  return {
```

Comment:
Could we reuse `newModel` function here?

Reply:
I don't think it's a good idea in this case, as this would make the newModel less understandable, as so far, newModel is for creating a new model based on an existing AlertQuery and a particular data source.
In here, it's something different, we create a new model for default data source and we are not relaying on an existing AlertQuery